.github/workflows/build.yml

```diff
@@ -37,11 +37,10 @@ jobs:
           ./.githooks/check-docs
         shell: bash
 
-  build:
-    name: Build
-    needs:
-      - check_format
-      - check_docs
```

Comment:
I think we can do this in parallel, and there is no reason to wait.
When I switch to pre-commit hooks completely, there will be no check_format at all and it will be even in a different workflow file

Reply:
There is no need to clog the CI and build anything if we don't even pass the pre-commit hooks, that's why we are waiting for these to succeed first.

Reply:
Got your point, reverted this change